packages/class/src/bmi.ts

```diff
@@ -0,0 +1,137 @@
+import { CLASS } from "./class";
+import { classConfig, ClassConfig } from "./config";
+
+/**
+ * A lightweight [BMI](https://bmi.readthedocs.io) like interface for the CLASS model.
+ *
+ * Inspiration https://github.com/uihilab/BMI-JS/blob/main/bmijs/bmi.js
+ *
+ * Deviations from the BMI standard
+ * - accessors do not use dest argument
+ * - initialize() accepts object instead of string
+ * - parameters() returns default config
+ */
+interface BmiLight<Config> {
+  initialize(config: Config): void;
+  update(): void;
+  get_component_name(): string;
+  get_output_item_count(): number;
+  get_output_var_names(): string[];
+  get_var_grid(name: string): number;
+  get_var_type(name: string): string;
+  get_var_location(name: string): string;
+  get_current_time(): number;
+  get_end_time(): number;
+  get_time_units(): string;
+  get_time_step(): number;
+  get_value(name: string): number[];
+  get_grid_type(): string;
+}
+
+const ouput_var_names: string[] = ["h", "theta", "dtheta", "q", "dq"] as const;
+
+export class BmiClass implements BmiLight<ClassConfig> {
+  config: ClassConfig = classConfig.parse({});
+  model: CLASS = new CLASS(this.config);
+
+  constructor() {
+    // no-op
```

Comment:
Could you explain why this no-op constructor is needed?

Reply:
I am sure it was needed before, but app now seems to work without it. I will get rid of it.